Packages/Blockchain/Tests/BlockchainTests/EthereumFeeServiceTests.swift

```diff
@@ -0,0 +1,92 @@
+// Copyright (c). Gem Wallet. All rights reserved.
+
+import Foundation
+import Testing
+import SwiftHTTPClient
+import Primitives
+import GemstonePrimitives
+import BigInt
+@testable import Blockchain
+
+struct EthereumFeeServiceTests {
+    static let chain: EVMChain = .ethereum
+    static let provider: Provider<EthereumProvider> = ProviderFactory.create(with: chain.chain.defaultBaseUrl)
+    static let config = GemstoneConfig.shared.config(for: chain)
+    static let service = EthereumService(chain: chain, provider: provider)
+
+    @Test
+    func testCalculatePriorityFeesWith1BlockRewards() {
+        let rewards = [["0x3b9aca00", "0x77359400", "0xb2d05e00"]] // 1 2 3
+        let priorityFees = Self.service.calculatePriorityFees(rewards: rewards, config: Self.config)
+
+        let expectedFees: [FeePriority: BigInt] = [
+            .slow: 1_000_000_000.asBigInt,
+            .normal: 2_000_000_000.asBigInt,
+            .fast: 3_000_000_000.asBigInt
+        ]
+
+        #expect(priorityFees[.slow] == expectedFees[.slow])
+        #expect(priorityFees[.normal] == expectedFees[.normal])
+        #expect(priorityFees[.fast] == expectedFees[.fast])
+    }
+
+    @Test
+    func testCalculatePriorityFeesWith10BlockRewards() {
+        let rewards = [
```

Comment:
don't use hex values here,

either create `BigInt` and mock response from the RPC response